pool/paymentmgr.go

```diff
@@ -644,13 +730,51 @@ func (pm *PaymentMgr) payDividends(ctx context.Context, height uint32, treasuryA
 		return nil
 	}
 
+	if pm.isProcessing() {
+		log.Info("Waiting for in-progress payment processing to " +
+			"complete before starting another")
+		return nil
+	}
+
+	pm.setProcessing(true)
```

Comment:
Just do `.Lock()` here and defer `.Unlock()`

Reply:
not looking to hold the lock and have subsequent calls wait to execute.